cmd/secret/cmd/root_test.go

```diff
@@ -89,47 +120,67 @@ func TestRoot_Set(t *testing.T) {
 }
 
 func TestRoot_Get(t *testing.T) {
-	key := "key value"
+	key := "key-value"
 	value := "60OBdPOOkSOu6kn8ZuMuXtAPVrUEFkPREydDwY6+ip/LrAFaHSc="
 	path := "testFile.txt"
-	file, err := os.Create(path)
-	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
-	defer cancel()
+	t.Run("success", func(t *testing.T) {
+		file, err := os.Create(path)
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+		defer cancel()
 
-	require.NoError(t, err)
-	defer func() {
-		require.NoError(t, os.Remove(path))
-	}()
+		require.NoError(t, err)
+		defer func() {
+			require.NoError(t, os.Remove(path))
+		}()
 
-	defer func() {
-		require.NoError(t, file.Close())
-	}()
+		defer func() {
+			require.NoError(t, file.Close())
+		}()
 
-	fileTestData := make(map[string]string)
-	fileTestData[key] = value
-	require.NoError(t, json.NewEncoder(file).Encode(&fileTestData))
+		fileTestData := make(map[string]string)
+		fileTestData[key] = value
+		require.NoError(t, json.NewEncoder(file).Encode(&fileTestData))
 
-	r := New()
-	r.cmd.SetArgs([]string{"get", "--key", key, "--cipher-key", "ck", "--path", path})
-	executeErr := r.Execute(ctx)
-	require.NoError(t, executeErr)
+		r := New()
+		r.cmd.SetArgs([]string{"get", "--key", key, "--cipher-key", "ck", "--path", path})
+		executeErr := r.Execute(ctx)
+		require.NoError(t, executeErr)
 
-	testFile, err := os.Open(path)
-	require.NoError(t, err)
+		testFile, err := os.Open(path)
+		require.NoError(t, err)
 
-	defer func() {
-		require.NoError(t, testFile.Close())
-	}()
+		defer func() {
+			require.NoError(t, testFile.Close())
+		}()
 
-	fileData := make(map[string]string)
-	require.NoError(t, json.NewDecoder(testFile).Decode(&fileData))
-	var got string
-	require.Len(t, fileData, 1)
-	for _, value := range fileData {
-		got = value
-		break // we iterate one time to get first value
-	}
-	require.EqualValues(t, value, got)
+		fileData := make(map[string]string)
+		require.NoError(t, json.NewDecoder(testFile).Decode(&fileData))
+		var got string
+		require.Len(t, fileData, 1)
+		for _, value := range fileData {
+			got = value
+			break // we iterate one time to get first value
+		}
+		require.EqualValues(t, value, got)
+	})
+	t.Run("success after get redis command", func(t *testing.T) {
+		redisURL := "localhost:6379"
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+		defer cancel()
+
+		r := New()
+		b := bytes.NewBufferString("test")
+		r.cmd.SetArgs([]string{"set", "--key", key, "--value", "test value", "--cipher-key", "ck", "--redis-url", redisURL})
+		executeErr := r.Execute(ctx)
+		require.NoError(t, executeErr)
+
+		r.cmd.SetArgs([]string{"get", "--key", key, "--cipher-key", "ck", "--redis-url", redisURL})
+		err := r.Execute(ctx)
+		require.NoError(t, err)
+		out, err := ioutil.ReadAll(b)
+		require.NoError(t, err)
+		require.EqualValues(t, "test", string(out))
+	})
```

Comment:
Why do you create `b` with exited value and then check with the same value? I assume it should be used for getting input from cli command.